VideoWeb/VideoWeb/ClientApp/src/app/shared/force-play-video.directive.ts

```diff
@@ -0,0 +1,70 @@
+import { AfterViewInit, Directive, ElementRef, Input, OnChanges, OnInit, Renderer2, RendererFactory2, SimpleChanges } from '@angular/core';
+import { Logger } from '../services/logging/logger-base';
+
+@Directive({
+    selector: '[appForcePlayVideo]'
+})
+export class ForcePlayVideoDirective implements OnInit, OnChanges, AfterViewInit {
+    private readonly loggerPrefix = '[ForcePlayVideoDirective] -';
+    private renderer: Renderer2;
+    @Input() mute: boolean | null = null;
+    private unsubscribeFromMouseDownCallback: () => void;
+    private unsubscribeFromTouchStartCallback: () => void;
+
+    private get videoElement(): HTMLVideoElement {
+        return this.elementRef.nativeElement as HTMLVideoElement;
+    }
+
+    constructor(private elementRef: ElementRef, renderer2Factory: RendererFactory2, private logger: Logger) {
+        this.renderer = renderer2Factory.createRenderer(null, null);
+    }
+
+    ngOnInit(): void {
+        this.configureVideoElement();
+        this.addEventListeners();
+    }
+
+    ngAfterViewInit(): void {
+        this.videoElement.play();
+
+        setTimeout(() => {
```

Comment:
so we just attempt to play the video after 5 seconds once more? What happens if it fails to play it the second time?

Reply:
There isn't really much else we can do. This technically should not be used it is there as a back up. As the touch start /mouse down should be the one that matters.